### DISCUSSION TITLE
USD support and Hydra Delegate

### DISCUSSION THREAD

**Discussion by adro79** [bnpr/Malt#587] [Feedback & Feature Requests]
Hi, I just discovered this render framework thanks to a comment on [this video](https://www.youtube.com/watch?v=yNO9pdWIuv4).

Since it's a powerful project, it would be nice to see it integrated with other open-source projects rather than just Blender.

[OpenUSD](https://github.com/PixarAnimationStudios/OpenUSD) could provide universal integration to other DCCs like [Gaffer](https://www.gafferhq.org/), Maya, [Houdini](https://www.sidefx.com/products/houdini/solaris/), Nuke, Blender (via Hydra) and most of the applications that [support](https://openusd.org/release/usd_products.html) this industry-standard technology. Also, it would permit more studios and artists to use and support the Malt project.

I hope you consider this feature on a long run once Malt get more mature.

**Comment by rosstheblenderanimator** [comment]
I think It makes more sense to get full supports for geometry nodes as it comes with blender and similar things to Houdini.

**Reply by adro79** [reply]
I agree. I'm not asking for this feature to be developed right now, as USD is not yet fully supported everywhere, but I recommend them to stay tuned so the don't miss the train.

**Comment by pragma37** [comment]
> [OpenUSD](https://github.com/PixarAnimationStudios/OpenUSD) could provide universal integration to other DCCs like [Gaffer](https://www.gafferhq.org/), Maya, [Houdini](https://www.sidefx.com/products/houdini/solaris/), Nuke, Blender (via Hydra) and most of the applications that [support](https://openusd.org/release/usd_products.html) this industry-standard technology.

Does this hold true in practice?
Can you take any renderer with a Hydra delegate, plug it into any DCC with Hydra support and things "just work" (The DCC UI gets populated with the renderer's specific settings and node systems)?

My impression is that Hydra only covers the scene geometry reliably, and renderers end-up needing a DCC-specific plugin/integration anyway. But maybe that's wrong/outdated info?

**Reply by adro79** [reply]
Every delegate can have its own render settings, like enable denoiser or diffuse samples.

![image](https://learn.foundry.com/katana/5.0/Content/Resources/Images/ug/hydra/hydra_dels9.jpg)
_Image from [Katana Docs](https://learn.foundry.com/katana/5.0/Content/ug/using_hydra_viewer/hydra-render-delegates.html)_

For other custom features you could create the necessary schemas and procedurals.  

I'm not a developer so I don't know what it would need to create a hydra delegate, but you can take a look at other implementations like [Moonray](https://github.com/dreamworksanimation/hdMoonray), [Cycles](https://github.com/blender/cycles/tree/main/src/hydra), [Arnold](https://github.com/Autodesk/arnold-usd), [Gatling](https://github.com/pablode/gatling) or [3Delight](https://gitlab.com/3Delight/HydraNSI).

**Reply by pragma37** [reply]
Thanks for the insights.

Note that render settings are different from schemas, so supporting the former doesn't mean they support the latter.

The main issue is that all the Hydra hosts out there are private software (and hella expensive), so I don't really have an actionable path for doing further research, not to mention testing/supporting an hypothetical Malt Hydra delegate.
Blender is the only exception, but the Hydra support is still pretty bare-bones.

**Reply by Another52** [reply]
> The main issue is that all the Hydra hosts out there are private software (and hella expensive), so I don't really have an actionable path for doing further research, not to mention testing/supporting an hypothetical Malt Hydra delegate. Blender is the only exception, but the Hydra support is still pretty bare-bones.

Gaffer is free and open source I heard

**Reply by adro79** [reply]
> > The main issue is that all the Hydra hosts out there are private software (and hella expensive), so I don't really have an actionable path for doing further research, not to mention testing/supporting an hypothetical Malt Hydra delegate. Blender is the only exception, but the Hydra support is still pretty bare-bones.
> 
> Gaffer is free and open source I heard

If you want to test example scenes, usdview is enough. 
There's also [usdtweak](https://github.com/cpichard/usdtweak) which give you basic editing and it's open source. Another option is [ShapeFX Loki](https://shapefx.app/), free for non-commercial and more complete.

**Reply by Another52** [reply]
Looks like there's no interest in this. Well 🥲

**Reply by pragma37** [reply]
>Gaffer is free and open source I heard

Gaffer doesn't have Hydra support AFAIK.
https://www.gafferhq.org/documentation/1.5.8.0/search.html?q=hydra&check_keywords=yes&area=default

> There's also [usdtweak](https://github.com/cpichard/usdtweak) which give you basic editing and it's open source. Another option is [ShapeFX Loki](https://shapefx.app/), free for non-commercial and more complete.

usdtweak seems to integrate Hydra Storm, but doesn't seem to support arbitrary Hydra delegates.
ShapeFX does, though, and can be used for free.
Good reference, thanks!
